code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -564,7 +564,10 @@
 	if(QDELETED(src))
 		return FALSE
 
-	if(info.distance_travelled > range || info.current_turf == info.target_turf)
+	if(info.distance_travelled > range)
+		return FALSE
+
+	if(!info.distance_travelled == 0 && info.current_turf == info.target_turf )
```

Comment:
```
	if(!value == 0)
```


this line makes absolutely zero sense please rewrite the check so it can be _understood_